### PR TITLE
Backend/feat:driver-data-decryption

### PR DESCRIPTION
**Pull request by MustafaAsad198**
## Type of Change

- [ ] Bugfix
- [x] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates

## Description

This is the Dashboard API that takes the list of driverID and encrypted driverData, and provides with a response of similar list  which contains decrypted driverData.

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## How did you test it?
image

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added integration tests for my changes where possible
- [ ] I tested the changes using integration tests
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable
- [ ] No leak detected in leakcanary

### CODE REVIEW

File: Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Management/Driver.hs [hunk 1]
```diff
@@ -315,3 +318,12 @@ getDriverSecurityDepositStatus :: (ShortId DM.Merchant -> City.City -> ApiTokenI
 getDriverSecurityDepositStatus merchantShortId opCity apiTokenInfo driverId mbServiceName = do
   checkedMerchantId <- merchantCityAccessCheck merchantShortId apiTokenInfo.merchant.shortId opCity apiTokenInfo.city
   Client.callDriverOfferBPPOperations checkedMerchantId opCity (.driverDSL.getDriverSecurityDepositStatus) driverId mbServiceName
+
```
Comment (vijaygupta18) [inline Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Management/Driver.hs:321]: can you can put this at the same place as other instances?

File: Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Management/Driver.hs [hunk 2]
```diff
@@ -315,3 +318,12 @@ getDriverSecurityDepositStatus :: (ShortId DM.Merchant -> City.City -> ApiTokenI
...
+instance Kernel.Types.HideSecrets.HideSecrets [Common.DriverEncDataReq] where
```
Comment (vijaygupta18) [inline Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Management/Driver.hs:322]: can you can put this instance at the same place as other instances?

Comment (vijaygupta18) [inline Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Management/Driver.hs:322]: can you can put this instance at the same place as other instances?

**Review by vijaygupta18 — CHANGES_REQUESTED** [review]

File: Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Management/Driver.hs (hunk 2 above)
Comment (MustafaAsad198) [reply Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Management/Driver.hs]: just did, can you review it?

File: Backend/app/dashboard/CommonAPIs/spec/ProviderPlatform/Management/API/Driver.yaml [hunk 3]
```diff
@@ -43,6 +43,16 @@ apis:
...
       migrate:
         endpointV2:
+  - POST: # PersonMobileNoAPI
+      endpoint: /driverDataDecryption
+      auth: ApiAuth DRIVER_OFFER_BPP_MANAGEMENT DRIVERS DRIVER_DECRYPTION
+      request:
+        type: "[DriverEncDataReq]"
+      response:
+        type: "[DriverDecDataResp]"
+      migrate:
```
Comment (hkmangla) [inline Backend/app/dashboard/CommonAPIs/spec/ProviderPlatform/Management/API/Driver.yaml:53]: You can remove migrate part.

**Review by hkmangla — CHANGES_REQUESTED** [review]

File: Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Management/Driver.hs [hunk 4]
```diff
@@ -11,6 +11,7 @@
 
  the GNU Affero General Public License along with this program. If not, see <https://www.gnu.org/licenses/>.
 -}
+{-# OPTIONS_GHC -Wno-orphans #-}
```
Comment (vijaygupta18) [inline Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Management/Driver.hs:14]: you can remove this too

**Review by vijaygupta18 — APPROVED** [review]

File: Backend/app/dashboard/CommonAPIs/spec/ProviderPlatform/Management/API/Driver.yaml (hunk 3 above)
Comment (MustafaAsad198) [reply Backend/app/dashboard/CommonAPIs/spec/ProviderPlatform/Management/API/Driver.yaml]: done

File: Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Management/Driver.hs (hunk 4 above)
Comment (MustafaAsad198) [reply Backend/app/dashboard/provider-dashboard/src/Domain/Action/ProviderPlatform/Management/Driver.hs]: done

**Review by vijaygupta18 — APPROVED** [review]

**Review by hkmangla — APPROVED** [review]

**Pull request merged**